tools/src/main/scala/scala/scalanative/codegen/Metadata.scala

```diff
@@ -6,7 +6,11 @@ import scalanative.nir._
 import scalanative.sema._
 import scalanative.linker.{Trait, Struct, Class}
 
-class Metadata(val linked: linker.Result, proxies: Seq[Defn]) {
+import scala.scalanative.build.TargetArchitecture
+
+class Metadata(val linked: linker.Result,
+               proxies: Seq[Defn],
+               val targetArchitecture: TargetArchitecture) {
```

Comment:
It looks like `targetArchitecture` is not used here.

Reply:
It's used when calculating field layouts, as `meta.targetArchitecture`.